dartagnan/src/main/java/com/dat3m/dartagnan/wmm/analysis/RelationAnalysis.java

```diff
@@ -723,8 +755,11 @@ public Knowledge visitCoherence(Relation rel) {
                     continue;
                 }
                 for (MemEvent w2 : nonInitWrites) {
+                    if (w1.is(PTX.WEAK) && w2.is(PTX.WEAK)) {
```

Comment:
Add a comment explaining why

Reply:
Added

Reply:
Now that I think again, this sound wrong to me. The comment clearly states "if they are not causally related", which means that two weak stores can indeed be in `co` (if they additionally are causally related). By skipping this pair from the may set computation, you are saying that they will never be related.

I also noted that in the encoding visitor of coherence, you still force one of the orders to happen.

I am still not sure what is the best solution (specially since AFAIK we have no litmus showing that `co` not being total is required). Does any test break if you remove the if check in `RelationAnalysis`?

Reply:
No this part doesn't affect any test we have. I also think maybe we should keep it when reading it. For example two weak writes of the same thread are in the cause-base relation (`cause-base = (po*; (sw; po*)+) | po`), then they shouldn't be skipped by visitCoherence here.

Reply:
You can only know dynamically (i.e. when the smt solver is doing the exploration) if the events are in causal order or not. Because of this, you can only know dynamically if they should or should not be in `co`. Putting differently they are executions where they are related by coherence. 

The may set computation is done statically. If you do not add a pair to the may set, it must be the case that in every possible execution the pairs is not in `co`. This invariant does not hold and thus is simply wrong to avoid adding the pair.

Reply:
Okay, I'll keep the weak writes of Coherence.